webapp/src/components/Analysis/PerturbationTestingPreview.test.tsx

```diff
@@ -0,0 +1,167 @@
+import { fireEvent, screen, waitFor } from "@testing-library/react";
+import { renderWithRouterAndRedux } from "mocks/utils";
+import PerturbationTestingPreview from "./PerturbationTestingPreview";
+import { AvailableDatasetSplits } from "types/api";
+import { setupServer } from "msw/node";
+import {
+  getPerturbationResponse,
+  getPerturbationResponseWithoutTrainFailureRate,
+  getPerturbationResponseWithFailureResponse,
+} from "mocks/api/mockPertubationAPI";
+
+const renderPerturbationTestingPreview = (
+  availableDatasetSplits: AvailableDatasetSplits
+) =>
+  renderWithRouterAndRedux(
+    <PerturbationTestingPreview
+      jobId="local"
+      pipeline={{ pipelineIndex: 0 }}
+      availableDatasetSplits={availableDatasetSplits}
+    />,
+    { route: "/local?pipeline_index=0", path: "/:jobId?:pipeline" }
+  );
+
+describe("PerturbationTestingPreview", () => {
+  const handlers = [getPerturbationResponse];
+  const server = setupServer(...handlers);
+
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+
+  it("should have two toggle buttons for the train and test", async () => {
+    renderPerturbationTestingPreview({ train: true, eval: true });
+    await waitFor(() => {
+      expect(screen.getAllByRole("button")).toHaveLength(2);
+    });
```

Comment:
No need to `waitFor` data.
```suggestion
    expect(screen.getAllByRole("button")).toHaveLength(2);
```